beetsplug/fish.py

```diff
@@ -89,14 +95,22 @@ def run(self, lib, opts, args):
         # If specified, also collect the values for these fields.
         # Make a giant string of all the above, formatted in a way that
         # allows Fish to do tab completion for the `beet` command.
-        home_dir = os.path.expanduser("~")
-        completion_dir = os.path.join(home_dir, '.config/fish/completions')
-        try:
-            os.makedirs(completion_dir)
-        except OSError:
-            if not os.path.isdir(completion_dir):
-                raise
-        completion_file_path = os.path.join(completion_dir, 'beet.fish')
+
+        completion_file_path = opts.output
+        if completion_file_path is not None:
+            completion_dir = os.path.dirname(completion_file_path)
+        else:
+            home_dir = os.path.expanduser("~")
+            completion_dir = os.path.join(home_dir, '.config/fish/completions')
```

Comment:
Maybe it would be easier to just make `~/.config/fish/completions/beet.fish` the actual `default` parameter to `add_option` above, and then stick to `completion_dir = os.path.dirname(...)` regardless of whether the parameter is set? As an added bonus, this would let the parameter include `~`, because we'd need to include `expanduser` on the entire path.